### ISSUE TITLE
sidebar behavior on tab selection

### ISSUE THREAD

**Issue by kipserv717** [ayushsharma82/ESP-DASH#203]
**Is your feature request related to a problem? Please describe.**
more of a ui annoyance, when selecting a tab on mobile device you have to manually close the sidebar

**Describe the solution you'd like**
close the sidebar when tab is selected

**Additional context**
ESP-DASH-Pro-v4-3-0
Android 14/Chrome 120.0.6099.231

**Comment by mathieucarbou** [comment]
I noticed the same issue. I did a lot of improvements on the pro version that I hope @ayushsharma82 will grab in future releases. 

To fix this issue, I've replaced in `portal/src/lib/Navbar.svelte`:

`<div class={`relative z-50 md:hidden ${hide ? 'hidden' : ''}`} role="dialog" aria-modal="true">`

with

`<div class={`relative z-50 lg:hidden ${hide ? 'hidden' : ''}`} role="dialog" aria-modal="true">`

you need to rebuild the UI with `npm run build` while being in the portal folder once done.

**Comment by mathieucarbou** [comment]
Sorry - I've realised this is not fixing the same issue ;-)

**Comment by mathieucarbou** [comment]
In the same file, juste update this function to add the `document.body.addEventListener()` and `document.body.removeEventListener()` calls

```js
const toggleNavbar = () => {
    if (collapsed === false) {
        collapsed = true;
        document.body.removeEventListener('click', toggleNavbar)
        setTimeout(() => {
            hide = true;
        }, 300);
    } else {
        hide = false;
        document.body.addEventListener('click', toggleNavbar)
        setTimeout(() => {
            collapsed = false;
        }, 100);
    }
}
```

**Comment by ayushsharma82** [comment]
@mathieucarbou I’ll grab those don’t worry 😂. As I said - in February when I’m getting enough time to review, merge, document & release everything.

**Comment by ayushsharma82** [comment]
Merged with v4.0.4 release of ESP-DASH Lite and v4.4.0 release of [ESP-DASH Pro](http).
